### PR TITLE
fix(Search): add default placeholder (a11y)

### PR DESCRIPTION
**Pull request by jsehull**
#12360 

Adding temporary solution as suggested in #12360 until greater decision is made.

#### Changelog

**New**

- add default placeholder value to `Search`

**Changed**

- n/a

**Removed**

- n/a

#### Testing / Reviewing

Visit Storybook > Search to confirm:
1. default placeholder is "search" when no placeholder is given
2. if a placeholder IS used, the placeholder is visible and NOT "search"
   - Search Playground is a good place for this

### CODE REVIEW

File: packages/react/src/components/Search/Search.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ const Search = React.forwardRef(function Search(
     onClear = () => {},
     onKeyDown,
     onExpand,
-    placeholder = '',
+    placeholder = 'search',
```
Comment (andreancardona) [inline packages/react/src/components/Search/Search.js]:
```suggestion
    placeholder = 'Search',
```

Comment (andreancardona) [reply packages/react/src/components/Search/Search.js]: Small suggestion - seems like custom is to capitalize - was browsing different examples:  http://web-accessibility.carnegiemuseums.org/code/search/

All else looks good 🔥

Comment (jsehull) [reply packages/react/src/components/Search/Search.js]: Good eye, can do!

**Review by tw15egan — APPROVED** [review]

**Comment by francinelucca** [comment]
@mbgower is this what you meant by "make the placeholder text in any search input into its name, with "search" as a fall back if no placeholder text exists." ? or do we need to add the `name` html attribute to the Input tag? I seem to have confused myself here

**Comment by mbgower** [comment]
HI @francinelucca, I find I need to navigate to the Search component when I follow the react build link (so if I'm doing that wrong, let me know!).

I see a default value in the search edit called "A default value". I want to make sure that no one thinks this is what I'm talking about (and I don't understand why that string exists, as I've never encountered such a thing in the real world).
image

If I clear this default value, then I see the placeholder text for this search input, which is "search"
image

In my experience, most teams do not use "Search" as their placeholder value, but this could be the fall back in Carbon where a team doesn't specify anything.  All Carbon needs to do is make it so whatever the string for the placeholder is, is also the (non-visible) label for the region. Remember, we already have region of _type_ search, so it's not like there isn't already context provided.

Does that answer your question?

**Comment by tw15egan** [comment]
I just pushed up some changes that @francinelucca and I were messing around with. It will set the placeholder value as the `aria-label` on the outer `div` that has a role of `search`. It seems like this way, we can have multiple Search instances with the same `labelText` as long as they have differing `placeholder` text. The screenreader reads the placeholder text, although it is still a bit verbose. 

Check the `with layer` stories and see if you come across any violations, and test with a screenreader

**Comment by francinelucca** [comment]
> I just pushed up some changes that @francinelucca and I were messing around with. It will set the placeholder value as the `aria-label` on the outer `div` that has a role of `search`. It seems like this way, we can have multiple Search instances with the same `labelText` as long as they have differing `placeholder` text. The screenreader reads the placeholder text, although it is still a bit verbose.
> 
> Check the `with layer` stories and see if you come across any violations, and test with a screenreader

Thanks @tw15egan , @mbgower!
@mbgower is this what you would expect screenreader-wise in these instances with the corrected changes in place? https://deploy-preview-13306--carbon-components-react.netlify.app/?path=/story/components-search--with-layer

**Comment by mbgower** [comment]
There is a slight inconsistent oddity on the first (and mabye third?) input not happening on the second one. But I think either is still an improvement.
I made a short video that hopefully demonstrates.

https://user-images.githubusercontent.com/14298245/224431887-b4fc3dff-e5a5-4413-bd5a-5cd890059688.mov

**Review by francinelucca — APPROVED** [review]

**Review by tw15egan — APPROVED** [review]

**Pull request merged**